client/src/components/History/Content/model/states.js

```diff
@@ -85,3 +85,6 @@ export const STATES = {
  * This list is ordered from highest to lowest priority. If any element is in error state the whole collection should be in error.
  */
 export const HIERARCHICAL_COLLECTION_JOB_STATES = ["error", "failed", "upload", "paused", "running", "queued", "new"];
+
+/** Datalist recommended states for the advanced history panel search */
+export const COMMON_HISTORY_PANEL_STATES = ["ok", "error", "failed", "upload", "paused", "running", "queued", "new"];
```

Comment:
Can you use https://github.com/mvdbeek/galaxy/blob/9c77e4d34bee5e023010dcbc3aeaf3f7e67db367/client/src/schema/schema.ts#L7326 ? That list should be the source of truth and will be updated if the backend changes.

Reply:
Yes, I can use that instead

Reply:
(You probably can only use that as the type annotation, I don't think direct re-use is possible since the interface is compiled away)

Reply:
Yes, I am trying it out and importing it into `HistoryFilters` which is a non-typescript component, and I am not being able to use it as a JS array

Reply:
Also, keep in mind that these are **Common** states that show up in a `<datalist>` that are not the only values users can enter into the field. The actual state filter field is a simple `<input>` field that takes any input

Reply:
Right, we'd have to make this a typescript file, we can do that another time. That said this is somewhat redundant with STATES, any reason not to use that (any maybe exclude `failed` as that's not a dataset state) ?

Reply:
> Also, keep in mind that these are **Common** states that show up in a `<datalist>` that are not the only values users can enter into the field. The actual state filter field is a simple `<input>` field that takes any input

What's the reason for that ? Excluding metadata_failed, discarded, empty seems very arbitrary ?

Reply:
> What's the reason for that ? Excluding metadata_failed, discarded, empty seems very arbitrary ?

The original ticket #15569 was asking a more sensible list of states for end-users who may not understand what states like `failed-metadata` etc. mean, so we have a more concise list of states in the `datalist` and have a button next to the field which opens up a modal showing all states and their info as well

Reply:
I mean, failed isn't a dataset state, @bernt-matthias actually knows what metadata_failed is, and the question mark there is an excellent place to describe all states. I don't think making arbitrary choices of what is an "important" state helps fix that issue and only makes it worse IMO.

Reply:
Ok, so what i can do is revert the datalist to what we had before, i.e.: ALL states from states.js: ![image](https://user-images.githubusercontent.com/78516064/223840240-bb518750-16c6-441e-b175-50a71c15d4d5.png) 

And ofc we then have the question mark button that explains them to the user ?

Reply:
Yeah, I think that'd be great. failed, empty and upload are maybe questionable (I don't think we're still using the upload and empty state, and failed isn't a dataset state)

Reply:
I would be super happy with included docs / definitions of the states that is accessible via the question mark. 

Looking forward to see the actual definitions of the states.